azkaban-common/src/main/java/azkaban/metric/MetricReportManager.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2012 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.metric;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.apache.log4j.Logger;
+
+
+/**
+ * Manager for access or updating metric related functionality of Azkaban
+ * MetricManager is responsible all handling all action requests from statsServlet in Exec server
+ * <p> Metric Manager 'has a' relationship with :-
+ * <ul>
+ * <li>all the metric Azkaban is tracking</li>
+ * <li>all the emitters Azkaban is supposed to report metrics</li>
+ * </ul></p>
+ */
+public class MetricReportManager {
+  /**
+   * Maximum number of metrics reporting threads
+   */
+  private static final int MAX_EMITTER_THREADS = 4;
+  private static final Logger _logger = Logger.getLogger(MetricReportManager.class);
+
+  /**
+   * List of all the metrics that Azkaban is tracking
+   * Manager is not concerned with type of metric as long as it honors IMetric contracts
+   */
+  private List<IMetric<?>> _metrics;
+
+  /**
+   * List of all the emitter listening all the metrics
+   * Manager is not concerned with how emitter is reporting value.
+   * Manager is only responsible to notify all emitters whenever an IMetric wants to be notified
+   */
+  private List<IMetricEmitter> _metricEmitters;
+  private ExecutorService _executorService;
+  // Singleton variable
+  private static volatile MetricReportManager _instance = null;
+  private static boolean _isManagerEnabled;
```

Comment:
Seems like this needs to be volatile for visibility purpose.  Different threads can call disableManager() and  isAvailable() methods